hoot-core/src/main/cpp/hoot/core/io/schema/LongIntegerFieldDefinition.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2015 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+#include "LongIntegerFieldDefinition.h"
+
+// hoot
+#include <hoot/core/util/Log.h>
+
+// Standard
+#include <limits>
+#include <sstream>
+
+namespace hoot
+{
+using namespace std;
+
+LongIntegerFieldDefinition::LongIntegerFieldDefinition()
+{
+  _defaultValue = -numeric_limits<long long>::max();
+  _min = -numeric_limits<long long>::max();
```

Comment:
Do you know why we are setting _min as -numeric_limits<long long>::max(), instead of ::min()?

Strictly speaking, for a signed, n-bit, twos-compliment number, max is going to be 2^(n-1)-1, and min is going to be 2^(n-1), so numeric_limits<>::min() is not the same as -numeric_limits::max()

I mean it's probably this way for a reason... I'm just curious if anyone knows what the reason is.